packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useUploadFileEntities.ts

```diff
@@ -120,18 +99,74 @@ export function useUploadFileEntities(
   onStorageLimitExceeded: () => void = noop,
 ): UseUploadFileEntitiesReturn {
   /**
-   * General flow for how the functions in this hook are used is
+   * General flow of the internal logic in this hook is
    *
    * 1. `initiateUpload(files)` - called by the caller of the hook
-   * 2. `willUploadsExceedLimit(files, usage)` - client-side check if the uploads will exceed the storage limit
-   * 3. `prepareDirsForUpload(files)` - sets up the folder paths, checks for existing entities with the same name
-   * 4. `postPrepareUpload(files)` - may prompt the user to confirm uploading new versions of files, based on the results of `prepareUpload`
-   * 5. `startUpload(files)` - starts tracking file uploads, for each file, calls `uploadPreparedFile`
-   * 6. `uploadPreparedFile(file)` - uploads the file and creates/updates the entity
+   *    - Verify that the upload can proceed. Specifically:
+   *       1. Validate the upload destination
+   *       2. Check if the uploads will exceed a storage limit
+   *    - If either of these checks fail, do not proceed.
+   * 2. `prepareDirsForUpload(files)` - sets up the folder paths, checks for existing entities with the same name
+   * 3. `startUploadOrPromptForConfirmation(files)`
+   *     - For each file, check if the user should be prompted before the upload starts (e.g. if a new version will be created to update an existing file)
+   *     - If one or more prompts are required, they are added to the `activePrompts` array. Once the user resolves all prompts, the upload may start.
+   *     - If no prompts are required, the upload starts immediately.
+   * 6. `onUploadComplete(file, fileHandleId)` - using the file handle, creates or updates the FileEntity in Synapse
    */
 
   const { synapseClient } = useSynapseContext()
 
+  const { mutateAsync: createEntityWithNewFile } = useCreateEntity()
+  const { mutateAsync: updateEntityWithNewFile } = useUpdateEntity()
+
+  const onUploadComplete = useCallback(
```

Comment:
Delegated upload logic to new hook. Moved logic to this callback to create/update the entity when the upload is complete